src/service/url-replacements-impl.js

```diff
@@ -1053,21 +1053,13 @@ export class UrlReplacements {
    */
   collectUnwhitelistedVarsSync(element) {
     const url = element.getAttribute('src');
-    const vars = Object.create(null);
-    this.expandStringSync(url, /* opt_bindings */ undefined, vars);
-    const varNames = Object.keys(vars)
-        // Strip arguments from collect_vars to match whitelist api.
-        .map(key => {
-          const argsIndex = key.indexOf('(');
-          return argsIndex === -1 ? key : key.substring(0, argsIndex);
-        });
-
+    const macroNames = new Expander(this.variableSource_).getMacroNames(url);
     const whitelist = this.getWhitelistForElement_(element);
```

Comment:
nit: is it possible to pass the whitelist to `Expander` instance instead?

Reply:
I want to make sure I understand, do you want the expander to return (1) the macros that will not be expanded? Or (2) the macros that will be expanded (in url && in whitelist)?

Reply:
I thought we could pass `whitelist` to the `Expander()`. And then when calling `getMacroNames`, `this.variableSource_.getExpr()` could by default return the the macros that will be expanded. I feel that's better than filtering by the whitelist here.

Reply:
I think I understand.  I have written it so that getMacroNames will respect the whitelist but @choumx is looking for the values that will not be expanded here. That is why we return the whole list and then filter for names not in whitelist. Let me know if you have an idea for a better way

Reply:
Got it. However it still sounds like the logic should live under `Expander` instead of urlReplacement Service to me. 
Instead I would prefer removing the helper method here, and have `batch-json` called `new Expander(urlReplacment.getVariableSource())....`  instead. 

Let me know what you think. The change LGTM anyhow.

Reply:
I agree that this logic should probably be somewhere else. I was thinking it should be in `batch-json`. Either way, lets discuss and do this in a follow up PR.